nixos/modules/config/update-users-groups.pl

```diff
@@ -3,22 +3,25 @@
 use File::Slurp;
 use JSON;
 
-make_path("/var/lib/nixos", { mode => 0755 });
-
-
 # Keep track of deleted uids and gids.
 my $uidMapFile = "/var/lib/nixos/uid-map";
 my $uidMap = -e $uidMapFile ? decode_json(read_file($uidMapFile)) : {};
 
 my $gidMapFile = "/var/lib/nixos/gid-map";
 my $gidMap = -e $gidMapFile ? decode_json(read_file($gidMapFile)) : {};
 
+my $is_dry = $ENV{'NIXOS_ACTION'} // "" eq "dry-activate";
+make_path("/var/lib/nixos", { mode => 0755 }) unless $is_dry;
 
 sub updateFile {
     my ($path, $contents, $perms) = @_;
+    return if $is_dry;
     write_file($path, { atomic => 1, binmode => ':utf8', perms => $perms // 0644 }, $contents) or die;
 }
 
+sub nscdInvalidate {
+    system("nscd --invalidate " . $_[0]) unless $is_dry;
```

Comment:
Safer to pass arguments as a list

```suggestion
    system("nscd", "--invalidate ", $_[0]) unless $is_dry;
```